pkg/network/setup/podnic.go

```diff
@@ -171,6 +181,13 @@ func (l *podNIC) sortIPsBasedOnPrimaryIP(ipv4, ipv6 string) ([]string, error) {
 }
 
 func (l *podNIC) discoverAndStoreCache() error {
+	ifaceLink, err := link.DiscoverByNetwork(l.handler, l.vmi.Spec.Networks, l.vmiSpecNetwork.Name)
```

Comment:
Can we do this at the actual usage? I mean, when we actually use `podInterfaceName`.

Reply:
It is impossible because the pod-interface cache relay on the pod interface name.

If we do not make the discovery at this time, the cache will be created with the wrong name for legacy VMs.